specs/EnableInputPassthrough.md

```diff
@@ -0,0 +1,102 @@
+Different from visual hosting, user inputs are dispatched to the HWND owned by CoreWebView2Controller using EnableInputPassthrough, it is not required for developers to send user inputs from the app to the WebView2 control. It is not expected to use EnableInputPassthrough together with visual hosting.
+
+# Examples
+## Win32 C++
+```cpp
+HRESULT AppWindow::CreateControllerWithInputPassthrough()
+{
+      //! [CreateControllerWithOptions]
+    auto webViewEnvironment10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
+    if (!webViewEnvironment10)
+    {
+        FeatureNotAvailable();
+        return S_OK;
+    }
+
+    wil::com_ptr<ICoreWebView2ControllerOptions> options;
+    // The validation of parameters occurs when setting the properties.
+    HRESULT hr = webViewEnvironment4->CreateCoreWebView2ControllerOptions(options.GetAddressOf());
+    if (hr == E_INVALIDARG)
+    {
+        ShowFailure(hr, L"Unable to create WebView2 due to an invalid profile name.");
+        CloseAppWindow();
+        return S_OK;
+    }
+    CHECK_FAILURE(hr);
+
+    wil::com_ptr<ICoreWebView2ControllerOptions3> webView2ControllerOptions3;
+    if (SUCCEEDED(options->QueryInterface(IID_PPV_ARGS(&webView2ControllerOptions3))))
+    {
+        CHECK_FAILURE(webView2ControllerOptions3->put_EnableInputPassthrough(TRUE));
+    }
+
+    CHECK_FAILURE(webViewEnvironment10->CreateCoreWebView2ControllerWithOptions(
+        m_mainWindow, options.get(),
+        Callback<ICoreWebView2CreateCoreWebView2ControllerCompletedHandler>(
+            this, &AppWindow::OnCreateCoreWebView2ControllerCompleted)
+            .Get()));
+
+    return S_OK;
+}
+```
+
+### .NET, WinRT
+```c#
+CoreWebView2Environment _webViewEnvironment;
+WebViewCreationOptions _creationOptions;
+public CreateWebView2Controller(IntPtr parentWindow)
+{
+    CoreWebView2ControllerOptions controllerOptions = new CoreWebView2ControllerOptions();
+    controllerOptions.EnableInputPassthrough = _creationOptions.EnableInputPassthrough;
+
+    CoreWebView2Controller controller = null;
+
+    if (_creationOptions.entry == WebViewCreateEntry.CREATE_WITH_OPTION)
+    {
+        controller = await _webViewEnvironment.CreateCoreWebView2ControllerAsync(parentWindow, options);
+    }
+    else
+    {
+        controller = await _webViewEnvironment.CreateCoreWebView2ControllerAsync(parentWindow);
+    }
+
+    //...
+}
+```
+
+# API Details
+## Win32 C++
+```IDL
+interface ICoreWebView2StagingControllerOptions;
+
+interface ICoreWebView2StagingControllerOptions : IUnknown {
+  /// `EnableInputPassthrough` property is to enable/disable input passing through
+  /// the app before being delivered to the WebView2.
+  [propget] HRESULT EnableInputPassthrough([out, retval] BOOL* value);
```

Comment:
Is the term 'input passthrough' something that we've made up for this feature or is it a pre-existing term used elsewhere? If its new, we might consider something more explicit, like `AllowHostInputProcessing`. The term 'passthrough' here is sort of ambiguous to me whether the input is passing by the host app and delivered straight to the browser, or the input passes first through the host app and can be processed there.

Reply:
+1 - This is the name I was thinking of except `AllowHostInputHandling` either one is fine for me

Reply:
Agreed

Reply:
Modified the API name, thanks!